### PR TITLE
Added specific managers: all_objects and deleted_objects.

### PR DESCRIPTION
**Pull request by AndreasBackx**
See #58, a summary:

* `SafeDeleteMixin.objects` remains unchanged, it only shows undeleted models by default.
* `SafeDeleteMixin.all_objects` (`SafeDeleteAllManager`) shows both deleted and non-deleted models.
* `SafeDeleteMixin.deleted_objects` (`SafeDeleteDeletedManager`) shows deleted models.
* `SafeDeleteManager.all(show_deleted=False)` added and `SafeDeleteManager.all_with_deleted` and `SafeDeleteManager.deleted_only` are deprecated. `SafeDeleteManager.all(show_deleted=True)` should only be used for related managers because it creates a new queryset and therefore `select_related` and other builder methods (seems to be the most appropriate term here) are lost. See some more comments about this below.
* `DELETED_ONLY_VISIBLE` and `DELETED_VISIBLE` visibilities added.

In this PR I focused on the implementation of how django-permanent does managers as explained in #58. I did however briefly look at the initial reason #58 was created which was that deleted models could not be refreshed from the database using `refresh_from_db`. But I found it to be harder than initially thought. The `refresh_from_db` implementation does the following:

```python
Model.objects.filter(pk=1).get()
```

That is the eventual queryset it kind of ends up with. The whole method creates a queryset which it then uses `get` on and therefore relies on all of the methods it uses before getting the eventual model. The current implementation of `DELETED_VISIBLE_BY_FIELD` makes this impossible because `SafeDeleteManager.filter` and `SafeDeleteManager.get` creates a new queryset when called and so the previous information is lost. I haven't looked too much into fixing this specific issue but I have an idea while writing this. We could move the filter from `SafeDeleteManager.get_queryset` to the place where the lazy queryset is evaluated (I'd have to look when this is done). It would allow us to work with all of the objects in the Manager. It would also fix the issue where `SafeDeleteManager.all(show_deleted=True)` creates a new queryset.

Documentation will be added once the implementation is complete.

What are your thoughts?

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -21,5 +21,5 @@ deps =
     django-110: Django>=1.10,<1.11
 commands =
     flake8: flake8 safedelete --ignore=E501
-    django: coverage run --parallel-mode {toxinidir}/runtests.py
```
Comment (AndreasBackx) [inline tox.ini]: Forgot to mention this in the PR. I added the posargs variable in the tox setup, [more info](http://tox.readthedocs.io/en/latest/example/general.html).

**Comment by Gagaro** [comment]
Look's good to me, thanks @AndreasBackx . I'll merge once you'll have added the doc.

**Comment by AndreasBackx** [comment]
@Gagaro I'll have a look at whether I can get `all(show_deleted=True)`, `filter(pk={pk})`, and `get(pk={pk})` not return a new queryset like I mentioned. That way `refresh_from_db` can be implemented and those methods work as they were designed.

**Comment by AndreasBackx** [comment]
@Gagaro any update on this or #65?

**Comment by Gagaro** [comment]
I thought you were still working on #65 (because of your comments). Is it ready to merge?

**Comment by AndreasBackx** [comment]
@Gagaro I'll add everything to the docs for both branches this week hopefully and I'll ping you once I'm done. I was just looking whether I got the LGTM before adding docs to something that would eventually change.

**Comment by AndreasBackx** [comment]
Almost 2AM when I pushed this. I have to sleep and catch my train in 6 hours so I'm gonna leave this for tomorrow.

**Pull request closed without merging**